src/index.js

```diff
@@ -1,98 +1,25 @@
-  }
-  render(data) {
-    const sourceSelectBox = document.getElementById("newsSource");
-    data.sources.map(({ id }, index) => {
-      const selectOptions = document.createElement("option");
-      selectOptions[index] += selectOptions.text = id;
-      sourceSelectBox.appendChild(selectOptions);
-    });
-  }
-}
-class NewssourceSelectComponent extends ApiFetcher {
-  constructor(data) {
-    super();
-  }
-  async fetch(httpurl) {
-    return await super.fetch(httpurl);
-  }
-  render(data) {
-    const elementId = document.getElementById("newsDetails");
-    let returnHtml = "",
-      uniqueVal = [];
-    if (data.status === "error") {
-      returnHtml = `<div id="error">${data.message}</div>`;
-    } else {
-      data.articles.map(
-        (
-          { author, title, description, publishedAt, url, urlToImage },
-          index
-        ) => {
-          uniqueVal = index === 0 ? `<h1>${author}</h1>` : "";
-          returnHtml += `${uniqueVal}<div class="newsTitle">${title}</div><div class="newsDescription">${description}</div><div class="publishDate">${publishedAt}</div><div class="imageContainer"><a href="${url}" target="_blank"><img class ="imageLazy" data-lazy=${urlToImage} /></a></div>`;
-        }
-      );
-    }
-    elementId.innerHTML = returnHtml;
-  }
-}
-class NewsAppConent {
-  constructor() {
-    this.newssourceComponentClass = new NewssourceComponent(this);
-    this.newssourceSelectComponent = new NewssourceSelectComponent(this);
-  }
-  intialize() {
-    const sourceSelectBox = document.getElementById("newsSource");
-
-    sourceSelectBox.addEventListener("click", event => {
-      urlConstructor.newsCategory = event.target.value;
-    });
+import urlConstructor from "./config.json";
+import SourceFetcher from "./sourceFetch.js";
 
-    const newsBtn = document.getElementById("newsSourceBtn");
-    newsBtn.addEventListener("click", () => {
-      urlConstructor.urlConstructor.newsCategory = sourceSelectBox.value;
-      let url =
-        urlConstructor.urlConstructor.url +
-        urlConstructor.urlConstructor.newsCategory +
-        urlConstructor.urlConstructor.apiKey;
+let httpurl = `${urlConstructor.urlConstructor[0].apiURL}${
+  urlConstructor.urlConstructor[4].apiSources
+}${urlConstructor.urlConstructor[3].apiKey}`;
 
-      this.newssourceSelectComponent.fetch(url).then(value => {
-        this.newssourceSelectComponent.render(value);
-      });
-      import(/* webpackChunkName: "print" */ "./print.js").then(mod => {
-        console.log(mod);
-      });
-    });
-    this.newsSource = this.newssourceComponentClass.fetch().then(value => {
-      this.newssourceComponentClass.render(value);
-    });
+var handler = {
+  get: function(obj, prop) {
+      if(prop === 'get'){
+          const innerCall = new Proxy(obj, {
+            apply: function(target, thisarg, args){
+                console.log(target.fetch(...args));
+            }
+          })
+        innerCall(httpurl)
```

Comment:
indentation is broken because you do not use `;` in line 16 for some reason - hard to read such code.